lib/tandem/tandemSimulator.js

```diff
@@ -79,6 +84,13 @@ exports.make = function(config){
     basal: function(event){
       ensureTimestamp(event);
       if (currBasal != null) {
+        if(currBasal.deliveryType === 'temp') {
+          if (currTempBasal !== null) {
+            currBasal.percent = currTempBasal.percent;
```

Comment:
Check your use(s) of truthiness. All your examples coming through are non-zero percentages.